src/Symfony/Component/Uuid/Uuid.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Uuid;
+
+/**
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ */
+class Uuid implements \JsonSerializable
+{
+    public const TYPE_1 = UUID_TYPE_TIME;
+    public const TYPE_3 = UUID_TYPE_MD5;
+    public const TYPE_4 = UUID_TYPE_RANDOM;
+    public const TYPE_5 = UUID_TYPE_SHA1;
+
+    public const VARIANT_NCS = UUID_VARIANT_NCS;
+    public const VARIANT_DCE = UUID_VARIANT_DCE;
+    public const VARIANT_MICROSOFT = UUID_VARIANT_MICROSOFT;
+    public const VARIANT_OTHER = UUID_VARIANT_OTHER;
+
+    private $uuid;
+
+    public function __construct(string $uuid)
+    {
+        if (!uuid_is_valid($uuid)) {
+            throw new \InvalidArgumentException('The UUID is not valid.');
+        }
+
+        $this->uuid = $uuid;
+    }
+
+    public static function v1(): self
+    {
+        return new self(uuid_create(UUID_TYPE_TIME));
+    }
+
+    public static function v3(self $uuidNamespace, string $name): self
+    {
+        return new self(uuid_generate_md5($uuidNamespace->uuid, $name));
+    }
+
+    public static function v4(): self
+    {
+        return new self(uuid_create(UUID_TYPE_RANDOM));
+    }
+
+    public static function v5(self $uuidNamespace, string $name): self
+    {
+        return new self(uuid_generate_sha1($uuidNamespace->uuid, $name));
+    }
+
+    public static function fromBinary(string $uuidAsBinary): self
+    {
+        return new self(uuid_unparse($uuidAsBinary));
+    }
+
+    public function toBinary(): string
+    {
+        return uuid_parse($this->uuid);
+    }
+
+    public function isNull(): bool
+    {
+        return uuid_is_null($this->uuid);
+    }
+
+    public function equals(self $other): bool
+    {
+        return $this->uuid === $other->uuid;
+    }
+
+    public function compare(self $other): int
+    {
+        return uuid_compare($this->uuid, $other->uuid);
+    }
+
+    public function getType(): int
+    {
+        return uuid_type($this->uuid);
+    }
+
+    public function getVariant(): int
+    {
+        return uuid_variant($this->uuid);
+    }
+
+    public function getTime(): int
+    {
+        return uuid_time($this->uuid);
+    }
+
+    public function getMac(): string
+    {
+        return uuid_mac($this->uuid);
+    }
+
+    public function __toString()
+    {
+        return $this->uuid;
+    }
+
+    public function jsonSerialize(): string
+    {
+        return $this->uuid;
+    }
```

Comment:
should we maybe add `__serialize` and `__unserialize` ?

Reply:
Not sure, what for?

Reply:
When wanting to serialize this object, other than json ?

Reply:
Because of the smaller payload this would provide? What do you have in mind? I can't guess sorry :)